core-processor/src/main/java/io/micronaut/inject/utils/NativeElementsHelper.java

```diff
@@ -201,11 +201,11 @@ private void processClassHierarchy(C owner,
         if (excludeClass(classNode)) {
             return;
         }
+        reduce(owner, collectedMethods, getMethods(classNode), cache, false, false);
```

Comment:
Why do you need this change? I think it was tested multiple times, and this should work.